### PR TITLE
Fix the access rights to make the it working

### PR DESCRIPTION
**Pull request by psa-avat**
Brief summary of the changes
----------------------------
Only fix the dockerfile to be able to build XCSoar

Related issues and discussions
------------------------------

### CODE REVIEW

**Review by lordfolken — COMMENTED** [review]
Chmod can take n arguments. https://www.man7.org/linux/man-pages/man1/chmod.1.html
This change should not make a difference other than adding another layer inside the docker image slowing down io operations.

**Review by lordfolken — CHANGES_REQUESTED** [review]
Ok reviewing changes on the ski slope is less than ideal. My previous analysis is wrong.
Can you do the copy, chmod and execution in one run statement? One should keep the layers of docker images low in number. Also please change the commit message to start with docker: fix race cond. in permissions on install scripts

File: src/Kobo/Model.cpp [hunk 1]
```diff
@@ -96,3 +97,17 @@ GetKoboWifiInterface() noexcept
...
+  static char buffer[16];
+  if (!ReadFromFile("/dev/mmcblk0", 0x200, buffer, sizeof(buffer)))
+    buffer[0]=0 ;
+  
+  // trunc the string to 8 char
+  buffer[8]=0 ;
+
+  return buffer ;
+
+}
```
Comment (MaxKellermann) [inline src/Kobo/Model.cpp:113]: Unused (and undocumented) code.

**Comment by MaxKellermann** [comment]
This PR could have been so trivial if you only had added the definition and checks necessary for Clara 2E, then I could have easily merged it - but you chose to add a useless function, split it into several commits, and do a questionable Dockerfile change which is completely unrelated to Clara 2E support!

File: src/Kobo/Model.cpp (hunk 1 above)
Comment (psa-avat) [reply src/Kobo/Model.cpp:113]: I thought it was pushed on my branch
The code is used for debug in the system window.

**Comment by psa-avat** [comment]
> Chmod can take n arguments. https://www.man7.org/linux/man-pages/man1/chmod.1.html This change should not make a difference other than adding another layer inside the docker image slowing down io operations.

Sure but the orginial one was misplaced. The execution was before the chmod and it failed.

**Comment by psa-avat** [comment]
Sorry

  For me it is related, because I used it to compile. This is why I've done
a first commit with only the Dockerfile.
  As I've done it on a separate fork, I thought that the other changes will
be push on this new fork ( as I had'nt access to master).
  I pushed all my modifications now. The Clara 2E is working except
regarding to the wifi.
  I'll continue working on it, and now I'll only push a full version.


Patrick

Le mer. 28 déc. 2022 à 12:16, Max Kellermann ***@***.***> a
écrit :

**Comment by psa-avat** [comment]
> This PR could have been so trivial if you only had added the definition and checks necessary for Clara 2E, then I could have easily merged it - but you chose to add a useless function, split it into several commits, and do a questionable Dockerfile change which is completely unrelated to Clara 2E support!

Sorry 

  For me it is related, because I used it to compile. This is why I've done a first commit with only the Dockerfile.
  As I've done it on a separate fork, I thought that the other changes will be push on this new fork ( as I had'nt access to master).
  I pushed all my modifications now. The Clara 2E is working except regarding to the wifi. 
  I'll continue working on it, and now I'll only push a full version.
 

Patrick

**Comment by MaxKellermann** [comment]
This PR only gets more confusing if you add fixup commits. Please amend existing commits instead of adding fixup commits.

**Comment by psa-avat** [comment]
Hello

  Sorry, I'll work on my side and push only one when finish.
   I had not seen that my pull request will automatically group all my
pushes.
   Indeed I would like to test the pull process with the issue on the
Dockerfile.

  Reading your email I understood that all my pushes were added to the
request.

  Must I create a fork for each modification. One for the Dockerfile and on
for the kobo ?

 For the Clara 2e model, I just need to fix the WiFi. Everything else is
working.
 I want also to check the bluefly module I received yesterday before
pushing all the modifications.



Regards


Le sam. 31 déc. 2022, 07:05, Max Kellermann ***@***.***> a
écrit :

**Comment by MaxKellermann** [comment]
> Must I create a fork for each modification. One for the Dockerfile and on for the kobo ?

No. This sounds like you don't know anything about how git works, and your question doesn't even have anything to do with my request to amend commits instead of adding fixup commits.

Right now, your PR says: apply those 6 changes to XCSoar! Those 6 changes consist of:
- "Fix the access rights to make the it working" (huh, what is "it"??)
- "Add CLARA HD 2E as a new model"
- "fix battery for CLARA_2E" (fix ... what? What was wrong? Hey, this commit doesn't fix anything, it adds support for the Clara 2E and should have been part of the previous commit)
- "add CLARA_2E model" (hey, you already did that in the second commit!??)
- "Add CLARA_2E" (what, doing this AGAIN??)
- "docker: fix acces rights on install scripts" (but... the first commit was supposed to do that, or did it not??)

Independent of whether you know how to use git - do you understand that your suggestion of modifying XCSoar is rather confusing? If you don't understand that, no amount of git knowledge/experience will fix this pull request.

git is just a tool that lets you express your suggestion to change something in a technical way. Before you learn how to use git, you should be able to formulate your suggestion in a non-confusing way, with human language, without git. After that, learn git and convert this suggestion to a git pull request.

**Comment by psa-avat** [comment]
Hi,

  For sure I don't know git well.
  But for my modifications it's very clear.

  First, I had to fix an issue in the Dockerfile to be able to compile and
test XCSoar on my side. This is my first commit and the subject of my PR.

  Second : As I bought a new KOBO which is a Clara HD 2nd, I decided to
implement this model on XCSoar to make it available for everyone.
  I am still working on it and will make a new PR for that too.

  Is it possible to rollback my commits ? So I'll be able to make an only
new one

Right now, your PR says: apply those 6 changes to XCSoar! Those 6 changes
consist of:

   - "Fix the access rights to make the it working" (huh, what is "it"??)

              ==> this is really a fix. The dockerfile did not work because
of a chmod made too late.

   - "Add CLARA HD 2E as a new model"
   - "fix battery for CLARA_2E" (fix ... what? What was wrong? Hey, this
   commit doesn't fix anything, it adds support for the Clara 2E and should
   have been part of the previous commit)
   - "add CLARA_2E model" (hey, you already did that in the second
   commit!??)
   - "Add CLARA_2E" (what, doing this AGAIN??)

          ==> ok this is only one modification related to the insertion of
a new Model.


   - "docker: fix acces rights on install scripts" (but... the first commit
   was supposed to do that, or did it not??)

           ==> yes but someone asked me to change the comment to prefix it
by docker:





Patrick




Le sam. 31 déc. 2022 à 08:51, Max Kellermann ***@***.***> a
écrit :

**Comment by MaxKellermann** [comment]
> Is it possible to rollback my commits ?

Everything is possible with git. The best way to fix up a PR branch is to amend existing commits. The easiest way to do that is to use stgit.

> ==> yes but someone asked me to change the comment to prefix it by docker

Which you did NOT.
Instead of changing the commit message, you created a new commit with a new commit message.

Do you understand the difference?

**Comment by psa-avat** [comment]
Yes I understand the difference but there was also a demand for a
modification

Le sam. 31 déc. 2022, 11:19, Max Kellermann ***@***.***> a
écrit :

**Pull request closed without merging**